### PR TITLE
Waltz request reimbursement page

### LINKED ISSUE #49: Request reimbursement page
**Description**
We want to have a form to allow sponsors to request reimbursements. Create a request reimbursement page form (IMPORTANT: Don't follow the design of the figma, you can look at it for reference and for the fields we need but we should be using a built in form from shadcn). Instead feel free to look at the setup organization form for reference. 

image

**Acceptance Criteria**

Steps:

- Utilize shadcn components to create the form  
- Add required fields shown in figma

### PR DESCRIPTION
**Pull request by lukewaltz**
## Developer: Luke Waltz

Closes #49 

### Pull Request Summary

added transaction page with form to request a transaction

### Modifications

- added: \app\reimbursements\page.tsx (contains request reimbursement form)
- added: \components\ui\date-picker.tsx (shadcn date-picker component used in reimbursement form)
- run npm i - i had to import react-hook-form, zod, and radix-ui/react-icons

### Testing Considerations

- it works on npm run dev
- NOTE: there is an issue with the datepicker component's use within the form, I couldn't figure it out so please take a look.

### Pull Request Checklist

- [X] Code is neat, readable, and works
- [X] Comments are appropriate
- [X] The commit messages follows our [guidelines](https://h4i.notion.site/Conventional-Commits-593452ad1179489399ad3bd696ef772a)
- [X] The developer name is specified
- [X] The summary is completed
- [X] Assign reviewers

### Screenshots/Screencast

image
image

### CODE REVIEW

File: src/app/reimbursements/page.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          )}
+        />
+        <FormField
+          control={form.control}
+          name="transactionDate"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel className="block mb-2"> Transaction Date </FormLabel>
+              <FormControl>
+                <DatePicker {...field} />
```
Comment (s-iloo) [inline src/app/reimbursements/page.tsx]: Looks great, why don't you checkout the documentation on shadcn with this link https://ui.shadcn.com/docs/components/date-picker
I'm pretty sure if you scroll to the bottom you'll see an example of how they use the datepicker within a form. Pretty sure you're gonna have to hardcode it but should be similar if not identical to how they do it on the docs.

**Review by s-iloo — CHANGES_REQUESTED** [review]
Looks great, why don't you checkout the documentation on shadcn with this link https://ui.shadcn.com/docs/components/date-picker
I'm pretty sure if you scroll to the bottom you'll see an example of how they use the datepicker within a form. Pretty sure you're gonna have to hardcode it but should be similar if not identical to how they do it on the docs.

File: src/app/reimbursements/page.tsx (hunk 1 above)
Comment (lukewaltz) [reply src/app/reimbursements/page.tsx]: image
image

fixed. PR should be updated. Thanks for the help.

**Review by s-iloo — CHANGES_REQUESTED** [review]
So when I try to enter a transaction amount and then click submit, it doesn't let me submit and highlights the transaction amount field in red. Might be some problem with the types, maybe you have to make the type a string or something? See if you can reproduce the error on your end 

image

File: src/app/reimbursements/page.tsx (hunk 1 above)
Comment (lukewaltz) [reply src/app/reimbursements/page.tsx]: Hey Ian, I fixed the issue with the transaction amount submissions. PR is updated.

**Review by s-iloo — APPROVED** [review]
Looks great!

**Pull request merged**